Rakefile

```diff
@@ -87,6 +87,24 @@ def converted_history(markdown)
           normalize_bullets(markdown)))))
 end
 
+def siteify_file(file, front_matter = {})
+  abort "You seem to have misplaced your #{file} file. I can haz?" unless File.exists?(file)
+  contents = File.read(file)
+  title = contents.match(/\A# (.*)$/)[1]
+  output_file = file.sub(/\.markdown\z/, ".md").downcase
+  slug = File.basename(output_file, ".md")
+  front_matter = front_matter.merge({
+    "title"     => title,
+    "layout"    => "docs",
+    "permalink" => "/docs/#{slug}/",
+    "note"      => "This file is autogenerated. Edit /#{file} instead."
+  })
+  contents.gsub!(/\A# #{title}\n\n?/, "")
+  contents = converted_history(contents) if output_file == "history.md"
```

Comment:
I didn't mean for them to be contradictory – I meant I'd rather avoid the `if` if possible. Could we pass in the content? Or have a method which fetches the content of a page and have a `case` in that method? `when /\Ahistory/`, then convert, otherwise read file and return?